### ISSUE TITLE
Potential Compatibility Issue for LimeTextExplainer with Python 3.7 (includes workaround)

### ISSUE THREAD

**Issue by edkrueger** [marcotcr/lime#236]
When using python=3.7.0, lime==0.1.1.32 and scikit-learn==0.20.0, I get an error when I try to use LimeTextExplainer after building a piplined sklearn model.

For example in the doc's `Lime - basic usage, two class case.ipynb`, the pipeline is made as follows:

```
vectorizer = sklearn.feature_extraction.text.TfidfVectorizer(lowercase=False)
train_vectors = vectorizer.fit_transform(newsgroups_train.data)
test_vectors = vectorizer.transform(newsgroups_test.data)
rf = sklearn.ensemble.RandomForestClassifier(n_estimators=500)
rf.fit(train_vectors, newsgroups_train.target)
c = make_pipeline(vectorizer, rf)
```
.

When I run,
```
explainer = LimeTextExplainer(class_names=class_names)
exp = explainer.explain_instance(newsgroups_test.data[idx], c.predict_proba, num_features=6)
```
I get the error

```
TypeError: object of type 'NoneType' has no len()
```
.

On python=2.7.15, lime==0.1.1.32, scikit-learn==0.20.0, I don't get any error.

By looking in the source code for sklearn.feature_extraction.text.CountVectorizer, I found they were using the regex `r"(?u)\b\w\w+\b"` in their tokenizer and this allowed me a work around by doing

```
tokenizer = lambda doc: re.compile(r"(?u)\b\w\w+\b").findall(doc)
explainer = LimeTextExplainer(class_names=class_names, split_expression=tokenizer)
exp = explainer.explain_instance(newsgroups_test.data[idx], c.predict_proba, num_features=6)
```
.

Since its likely that many user will want to just use the default tokenizer with scikit-learn and lime.
I have some ideas to fix this, but I wanted to make sure y'all found this desirable before I looked into it further and made a PR.

Thanks!

**Comment by marcotcr** [comment]
Sorry for the delay in responding, I was away on vacation. Sure, it would be nice to get this fixed! 
Do you have a short example string for which the problem happens?

**Comment by mesterhd** [comment]
I can confirm the reported issue when I pass a sklearn pipeline (FunctionTransformer + CountVectorizer + TfidfTransformer + classifier) to LimeTextExplainer.explain_instance. I'm using Python 3.7.1, scikit-learn==0.20.1, and lime==0.1.1.32. I did not encounter any issues with Python 3.6.6. The proposed fix works for me.

**Comment by edkrueger** [comment]
Sorry for the delay. As far as I can tell, this happens with every string that I pass.

Here is the full stack trace:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-27-6e1cf88a9930> in <module>()
----> 1 exp = explainer.explain_instance("A", cv_lr.predict_proba, num_features=6)
      2 exp.show_in_notebook(text=True)

/anaconda3/envs/LP-NLP/lib/python3.7/site-packages/lime/lime_text.py in explain_instance(self, text_instance, classifier_fn, labels, top_labels, num_features, num_samples, distance_metric, model_regressor)
    384         indexed_string = IndexedCharacters(
    385             text_instance, bow=self.bow) if self.char_level else IndexedString(
--> 386             text_instance, bow=self.bow, split_expression=self.split_expression)
    387         domain_mapper = TextDomainMapper(indexed_string)
    388         data, yss, distances = self.__data_labels_distances(

/anaconda3/envs/LP-NLP/lib/python3.7/site-packages/lime/lime_text.py in __init__(self, raw_string, split_expression, bow)
    113         self.as_np = np.array(self.as_list)
    114         self.string_start = np.hstack(
--> 115             ([0], np.cumsum([len(x) for x in self.as_np[:-1]])))
    116         vocab = {}
    117         self.inverse_vocab = []

/anaconda3/envs/LP-NLP/lib/python3.7/site-packages/lime/lime_text.py in <listcomp>(.0)
    113         self.as_np = np.array(self.as_list)
    114         self.string_start = np.hstack(
--> 115             ([0], np.cumsum([len(x) for x in self.as_np[:-1]])))
    116         vocab = {}
    117         self.inverse_vocab = []

TypeError: object of type 'NoneType' has no len()
```

**Comment by marcotcr** [comment]
The problem is that python3.7 changed when you add ^ or $ to a regex. The proposed fix is a little different then the behavior we want, because it ignores 1 letter words, and also does not include non-words (such as punctuation, etc). 
Anyway, fixed in 9c906b4. Thanks!
